project_setup.py

```diff
@@ -0,0 +1,253 @@
+    except KeyError:
+        url_guess = MSG_GIT_URL_NOT_FOUND
+
+    # this could be a problem if someone accidentally hits enter and the URL is
+    # is set to MSG_GIT_URL_NOT_FOUND. Maybe change to a while loop if this
+    # becomes an issue.
+    repo_url = input_func(PROMPT_REMOTE_REPO.format(url_guess=url_guess)).strip()
+
+    return repo_url or url_guess
+
+
+def get_package_name(
+    repo_url:str,
+    input_func:Callable[[Optional[str]], str] = input
+) -> str:
+    """Gets the package name from the user.
+
+    Guesses the package name from the repo URL.
+
+    Args:
+        repo_url (str): The URL of the remote repository.
+        input_func (Callable[[Optional[str]], str], optional): The input function to use. Defaults to input.
+
+    Returns:
+        str: The package name.
+    """
+    repo_name = os.path.basename(repo_url).split('/')[-1].split('.')[0]
+    guessed_name = make_name_safe(repo_name)
+
+    input_name = input_func(PROMPT_PKG_NAME.format(name_guess=guessed_name)).strip()
+
+    return input_name or guessed_name
+
+
+def replace_file_contents(
+    new_str:str,
+    old_str:str,
+    file_path:str,
+) -> bool:
+    """Replaces instances of `old_str` with `new_str` in file at file_path.
+
+    Args:
+        old_str (str): The string to replace.
+        new_str (str): The string to replace `old_str` with.
+        file_path (str): The path to the file to modify.
+
+    Returns:
+        None
+    """
+    with open(file_path, "r") as f:
+        content = f.read()
+
+    new_content = content.replace(old_str, new_str)
+
+    if new_content != content:
+        with open(file_path, "w") as f:
+            f.write(new_content)
+        return True
+
+    return False
+
+
+def self_destruct(input_func:Callable[[str], str] = input) -> None:
+    """Deletes this file upon confirmation."""
+
+    if input_func("Delete this file? [y/N]: ").strip().lower() == "y":
+        os.remove(__file__)
+
+
+def rtd_project_guess(repo_url:str) -> str:
+    """Guesses the RTD project name from the repo URL."""
+    return repo_url.replace(".git", "").split('/')[-1]
+
+
+def codecov_project_guess(repo_url:str) -> str:
+    """Guesses the codecov project name from the repo URL."""
+    return "/".join(repo_url.replace(".git", "").split('/')[-2:])
+
+
+def run_setup(
+    repo_url:str,
+    package_name:str,
+    dir_path:str = ".",
+    input_func:Callable[[Optional[str]], str] = input,
+    output_func:Callable[[str], None] = print,
+) -> None:
+    # Update pyproject.toml and README.md, which contain the repo URL===========
+    rtd_guess = rtd_project_guess(repo_url)
+    codecov_guess = codecov_project_guess(repo_url)
+
+    replace_args = zip(
+        [repo_url, repo_url, rtd_guess, codecov_guess],
+        [TEMPLATE_REPO_URL, TEMPLATE_REPO_URL,  "ssec-jhu-base-template", "ssec-jhu/base-template"],
+        map(lambda f: os.path.join(dir_path, f), ["pyproject.toml", "README.md", "README.md", "README.md"]),
```

Comment:
not a problem of course, but maybe replace usage of `os.path.` everywhere with `pathlib`

```py
Path(dir_path, f)
```

or if dir path is a Path
```py
dir_path / f
```

Reply:
I use os.path out of habit, but pathlib is probably the right way to go. I'll change it.